### PR TITLE
Use server-side unread count

### PR DESCRIPTION
**Pull request by mspensieri**
@jugarrit @lemieux @alavers 

todo still: 
* bump smooch-core to `0.0.15`
* unit tests? (what should I test?)

### CODE REVIEW

**Comment by lemieux** [comment]
I'd maybe just test that smooch core gets called correctly whenever the reset count happens.

File: src/js/reducers/conversation-reducer.js [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import { ADD_MESSAGE, MESSAGES_CLEARED, SET_CONVERSATION, RESET_CONVERSATION } from 'actions/conversation-actions';
+import * as ConversationActions from 'actions/conversation-actions';
 import { RESET } from 'actions/common-actions';
 
 const INITIAL_STATE = {
```
Comment (lemieux) [inline src/js/reducers/conversation-reducer.js]: you should add the default unread count here

File: src/js/components/chat-input.jsx [hunk 2]
```diff
@@ -35,8 +36,8 @@ export class ChatInputComponent extends Component {
     }
 
     onFocus(e) {
-        if (getReadTimestamp() > 0) {
-            updateReadTimestamp(0);
+        if (store.getState().conversation.unreadCount > 0) {
```
Comment (mspensieri) [inline src/js/components/chat-input.jsx]: Am I correct in assuming this should be passed in through props?

Comment (Mario54) [inline src/js/components/chat-input.jsx]: Yes, you can add the `conversation` state in the `connect` call at the end of the file I think

Comment (mspensieri) [inline src/js/components/chat-input.jsx]: Hmm this makes it harder to unit test actually

Comment (lemieux) [inline src/js/components/chat-input.jsx]: If you need the component to rerender when that value changes, it needs to go in  the `connect` part at the bottom, if not, it doesn't matter.

**Comment by lemieux** [comment]
:+1:

**Pull request merged**